src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -543,6 +569,21 @@ def construct_payload(self):
             else:
                 set_managed_identity(self.cmd, self.get_argument_resource_group_name(), self.containerapp_def, user_assigned=[self.get_argument_registry_identity()])
 
+        if self.get_argument_source():
```

Comment:
I think you need to add the change to ContainerAppPeviewCreateDecorator. ContainerAppCreateDecorator should only contains GA code and preview features please go to ContainerAppPeviewCreateDecorator

Reply:
Since `source` and `repo` are not going to be in preview, do we still need to move this to `ContainerAppPeviewCreateDecorator`?

Reply:
Hi @snehapar9 
Yes, you need to add `is_preview=True`, because we are going to move existed GA features to azure-cli core in this PR: https://github.com/Azure/azure-cli/pull/27078
And all new arguments which only exist in extension should have `is_preview=True`, after the argument move into azure-cli, the arguments in extension could remove `is_preview=True` tag.

Reply:
Thanks for the clarification @Greedygre! However, looks like `ContainerAppPreviewCreateDecorator`'s [construct_payload](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/containerapp_decorator.py#L1195) method calls `ContainerAppCreateDecorator` methods [construct_payload](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/containerapp_decorator.py#L351) under the hood. 
Can you please confirm if we should handle the `source` and `repo` scenarios separately only within `ContainerAppPreviewCreateDecorator` classe's `construct_payload` method?

Reply:
Yes, we should handle the --source and --repo scenarios separately only within `ContainerAppPreviewCreateDecorator `class's `construct_payload` method.
You can refer to this parameter `--bind`
The argument `--bind` already existed in `ContainerAppPreviewCreateDecorator`, 
    ```
def construct_payload(self):
        super().construct_payload()
        self.set_up_service_bindings()
        self.set_up_unbind_service_bindings()
```

Reply:
Updated